jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -260,22 +263,40 @@ private Image pullBaseImage(
   }
 
   /**
-   * Looks through a manifest list for any amd64/linux manifest and downloads and returns the first
-   * manifest it finds.
+   * Looks through a manifest list for the user specified arch/os manifest and downloads and returns
+   * the first manifest it finds.
    */
-  private ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
+  @VisibleForTesting
+  ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
       RegistryClient registryClient, V22ManifestListTemplate manifestListTemplate)
       throws RegistryException, IOException {
 
-    List<String> digests = manifestListTemplate.getDigestsForPlatform("amd64", "linux");
+    Set<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
```

Comment:
I think we need to throw an error if there is more than one platform. Something along the line of

```
Preconditions.checkArgument(platforms.size() == 1, "multiple base image platforms is not yet support; please specify only one platform");
```

You can do this in a separate PR if you want. Regardless, you need to add a test.

Reply:
@chanseokoh I am going to add this ` check `  in this PR

Reply:
Let's do it in a separate PR. We may choose a different place for the check.

Reply:
Sorry, let's not* do it.

Reply:
Oh, wait. What I said was right. Let's not do it here but in a separate PR.

Reply:
Ok let me revert my commit

Reply:
Sounds good. Avoid significantly changing the original intention of a PR (e.g., adding a new feature that can be done in a separate PR) (unless necessary) when people started reviewing.

For the check, I'm thinking maybe we can check and raise an error at an earlier point. We should also add a test too anyway.

Reply:
Yes the feature requires a test or two to confirm its functionality .At which point ,do you think we could do the check?

Reply:
`JibContainerBuilder.containerize()` came to my mind, but I haven't given much thought actually. I'll need to look into the code, but in the meantime, you can continue work on other stuff.

Reply:
I think we can validate the input in `JibContainerBuilder.containerize()` and that `UnsupportedOperationException` could work. But if you think there's a better place, we can discuss.

```java
    try (BuildContext buildContext = toBuildContext(containerizer);
        TimerEventDispatcher ignored =
            new TimerEventDispatcher(
                buildContext.getEventHandlers(), containerizer.getDescription())) {
      if (...) {
        throw UnsupportedOperationException("... not yet supported ...");
      }
      logSources(buildContext.getEventHandlers());
```

Reply:
Single Platform Specification Support #2641